### PR TITLE
chore(common/registry): implement __repr__ for Registry

### PR DESCRIPTION
**Pull request by FateScript**
It's better for users to see what objects are in registry, so I implemented __repr__ for Registry.

Please kindly review :)

### CODE REVIEW

File: fvcore/common/registry.py [hunk 1]
```diff
@@ -73,3 +75,10 @@ def get(self, name: str) -> object:
 
     def __contains__(self, name: str) -> bool:
         return name in self._obj_map
+
+    def __repr__(self) -> str:
+        table_headers = ["Names", "Objects"]
+        table = tabulate(
+            self._obj_map.items(), headers=table_headers, tablefmt="fancy_grid"
+        )
+        return "Registry of {}:\n".format(self._name) + table
```
Comment (ppwwyyxx) [inline fvcore/common/registry.py:84]: could you add 

```
__str__ = __repr__
```

**Review by ppwwyyxx — COMMENTED** [review]
Thanks, could you show (in comments) what the result looks like?

**Comment by FateScript** [comment]
> Thanks, could you show (in comments) what the result looks like?

It looks like:
```Python3
In [2]: OPTIMIZERS
Out[2]:
Registry of optimizers:
╒════════════╤══════════════════════════════════════════════╕
│ Names      │ Objects                                      │
╞════════════╪══════════════════════════════════════════════╡
│ ASGD       │ <class 'torch.optim.asgd.ASGD'>              │
├────────────┼──────────────────────────────────────────────┤
│ Adadelta   │ <class 'torch.optim.adadelta.Adadelta'>      │
├────────────┼──────────────────────────────────────────────┤
│ Adagrad    │ <class 'torch.optim.adagrad.Adagrad'>        │
╘════════════╧══════════════════════════════════════════════╛
```

File: fvcore/common/registry.py (hunk 1 above)
Comment (FateScript) [reply fvcore/common/registry.py:84]: fixed.

**Pull request closed without merging**